src/webui/www/private/scripts/client.js

```diff
@@ -354,51 +354,52 @@ window.addEvent('load', function() {
     const serverState = {};
 
     const removeTorrentFromCategoryList = function(hash) {
-        if (hash === null || hash === "")
+        if (!hash)
             return false;
+
         let removed = false;
-        Object.each(category_list, function(category) {
-            if (Object.contains(category.torrents, hash)) {
-                removed = true;
-                category.torrents.splice(category.torrents.indexOf(hash), 1);
-            }
-        });
+        for (const key in category_list) {
```

Comment:
I may be mistaken, but `for-in` in IS may return something else besides the expected indexes. I remember we came across this earlier, and it required some kind of additional checks of `key`.

Reply:
Thanks! Added checking for it.

Reply:
Worth noting that `Object.each()` from mootools library would gather all entries into an array before processing them. So it won't be preferable when there are many elements.
https://github.com/mootools/mootools-core/blob/187a16bae2d7144ea4b81aa5c0586498f0fe17c7/Source/Core/Core.js#L345-L366

Also, ideally, js Object `{}` should be replaced by `Map()` whenever possible, but this is something for another time.